pkg/cmd/create/create.go

```diff
@@ -859,3 +879,156 @@ func defaultAttachFunc(o *CreateOptions, containerToAttach *corev1.Container, si
+		jobCompletionErrChan <- err
+	}()
+
+	select {
+	case sig := <-sigChan:
+		fmt.Fprintf(o.Out, "Received signal: %v, stopping the job and cleaning up...\n", sig)
+		err := k8sclient.BatchV1().Jobs(o.Namespace).Delete(ctx, jobName, metav1.DeleteOptions{
+			PropagationPolicy: ptr.To(metav1.DeletePropagationForeground),
+		})
+		if err != nil {
+			return err
+		}
+	case err := <-watcherErrChan:
+		if err != nil {
+			fmt.Fprintf(o.Out, "Error watching job pods: %v\n", err)
+			return err
+		}
+	case err := <-jobCompletionErrChan:
+		if err != nil {
+			fmt.Fprintf(o.Out, "Error checking job completion: %v\n", err)
+			return err
+		}
+		fmt.Fprint(o.Out, "Job logs streaming finished.")
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) watchJobPods(ctx context.Context, clientGetter util.ClientGetter, jobName string) error {
+	k8sclient, err := clientGetter.K8sClientset()
+	if err != nil {
+		return err
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("job-name=%s", jobName),
+	}
+
+	watcher, err := k8sclient.CoreV1().Pods(o.Namespace).Watch(ctx, listOptions)
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	for event := range watcher.ResultChan() {
+		pod, ok := event.Object.(*corev1.Pod)
+		if !ok {
+			continue
+		}
+
+		if event.Type == watch.Added || event.Type == watch.Modified {
+			if pod.Status.Phase == corev1.PodRunning {
+				go func() {
+					err = o.streamLogsFromPod(ctx, clientGetter, pod.Name)
+					if err != nil {
+						fmt.Fprintf(o.Out, "Error streaming logs from pod %q: %v\n", pod.Name, err)
+					}
+				}()
+			}
+		}
+	}
+	return nil
+}
+
+func (o *CreateOptions) streamLogsFromPod(ctx context.Context, clientGetter util.ClientGetter, podName string) error {
+	k8sclient, err := clientGetter.K8sClientset()
+	if err != nil {
+		return err
+	}
+
+	_, loaded := o.activeStreams.LoadOrStore(podName, true)
+	if loaded {
+		// pod is already streaming logs
+		return nil
+	}
+	defer o.activeStreams.Delete(podName)
+
+	fmt.Fprintf(o.Out, "Starting log streaming for pod %s...\n", podName)
+
+	logOptions := &corev1.PodLogOptions{
+		Follow: true,
+	}
+	req := k8sclient.CoreV1().Pods(o.Namespace).GetLogs(podName, logOptions)
+
+	logStream, err := req.Stream(ctx)
+	if err != nil {
+		return err
+	}
+	defer logStream.Close()
+
+	_, err = io.Copy(o.Out, logStream)
+	return err
+}
+
+func (o *CreateOptions) verifyJobFinished(ctx context.Context, clientset kubernetes.Interface, jobName string) error {
+	watchOptions := metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", jobName),
+	}
+
+	watcher, err := clientset.BatchV1().Jobs(o.Namespace).Watch(ctx, watchOptions)
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	for event := range watcher.ResultChan() {
+		job, ok := event.Object.(*batchv1.Job)
+		if !ok {
+			continue
+		}
+
+		if job.Spec.Completions != nil && job.Status.Succeeded == *job.Spec.Completions {
+			return nil
+		}
```

Comment:
```suggestion
		if job.Spec.Completions != nil && job.Status.Succeeded+job.Status.Failed == ptr.Deref(job.Spec.Completions, 0 {
			return nil
		}
```
Maybe something like this?

Reply:
I'm also wondering if we need this since we're already checking the conditions?